### PR TITLE
Allow empty text on Badge

### PR DESCRIPTION
**Pull request by faergeek**
See #76

### CODE REVIEW

File: src/Badge.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ class Badge extends React.Component {
 
         return React.cloneElement(element, {
             className: 'mdl-badge',
-            'data-badge': this.props.text
+            'data-badge': this.props.text || undefined
```
Comment (tleunen) [inline src/Badge.js]: why not just `this.props.text`?
Does mdl render something if it's an empty string?

File: demo/badge/badge.js [hunk 2]
```diff
@@ -2,9 +2,20 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import Icon from '../../src/Icon';
 import Badge from '../../src/Badge';
+import Textfield from '../../src/Textfield';
 
 class Demo extends React.Component {
+    state = {
```
Comment (tleunen) [inline demo/badge/badge.js]: state should be on the instance inside the constructor, not static ;)

Comment (faergeek) [inline demo/badge/badge.js]: It isn't static, it's on instance, static should be prepended by `static` keyword :-)

File: src/Badge.js (hunk 1 above)
Comment (faergeek) [inline src/Badge.js]: mdl has a selector that looks like this `.mdl-badge[data-badge]:after` so it renders badge if there's `data-badge` attribute even if it's empty.

And `undefined` does not render attribute at all. Check the DOM in devtools.

Comment (faergeek) [inline src/Badge.js]: We could set prop validation to `any` and convert `text` prop to string though.

Comment (faergeek) [inline src/Badge.js]: Sorry, actually we can't, undefined equals string 'undefined' if it's converted to string.

**Comment by faergeek** [comment]
No need to merge commit for demo, it's just for demonstration/testing.
I'm going to remove it if it's not needed anymore.

**Comment by tleunen** [comment]
So.. What you're trying to do is to not display the badge icon if the text is empty, right?

**Comment by faergeek** [comment]
Exactly.

Otherwise we would need to conditionally wrap/unwrap element.

**Comment by tleunen** [comment]
And React displays the attribute, even if `text` is empty?

**Comment by faergeek** [comment]
Attribute is rendered until it doesn't equal to `null` or `undefined`.

**Comment by tleunen** [comment]
Should be fixed now.

**Pull request closed without merging**